### PR TITLE
Documentation: Describe extra clang tools built alongside BuildClang.sh

### PR DESCRIPTION
**Pull request by ADKaster**
Also rephrase some of the information on updated clang-format binaries

### CODE REVIEW

**Review by supercomputer7 — APPROVED** [review]
Seems OK to me, even better now with the how you re-phrased some of my sentences in #16305 :)

File: Documentation/AdvancedBuildInstructions.md [hunk 1]
```diff
@@ -199,25 +199,35 @@ Alternatively, you may prefer to copy `Build/_disk_image` and `Build/Kernel/Kern
...
 
-There 3 options to acquire an updated clang-format tool:
-1) If you have a Debian-based (apt-based) distribution, refer to [SelfHostedRunners.md](https://github.com/SerenityOS/serenity/blob/master/Documentation/SelfHostedRunners.md) document to learn more about using the LLVM apt repositories to install the latest clang-format tool.
-2) Compile using `Toolchain/BuildClang.sh` as being pointed above use the compiled `Toolchain/Local/clang/bin/clang-format` binary.
-3) Compile LLVM from source as pointed [here](https://llvm.org/docs/GettingStarted.html#compiling-the-llvm-suite-source-code).
-    
+Some OS distributions don't ship bleeding-edge clang-format binaries. Below are 3 options to acquire an updated clang-format tool:
+
+1) If you have a Debian-based (apt-based) distribution, use the [LLVM apt repositories](https://apt.llvm.org) to install the latest release of clang-format.
+2) Compile the SerenityOS-patched LLVM from source using ``Toolchain/BuildClang.sh`` as described above and use the compiled ``Toolchain/Local/clang/bin/clang-format`` binary in your editor and terminal. The meta-lint-ci Pre-commit hook will automatically pick up the Toolchain clang-format binary.
```
Comment (bgianfo) [inline Documentation/AdvancedBuildInstructions.md:232]: Should we state which one "We" recommend?

**Review by bgianfo — APPROVED** [review]

File: Documentation/AdvancedBuildInstructions.md (hunk 1 above)
Comment (supercomputer7) [reply Documentation/AdvancedBuildInstructions.md:232]: Not really in my opinion. As long as it ensures the user has an updated clang-format, we don’t really care.

Comment (ADKaster) [reply Documentation/AdvancedBuildInstructions.md]: I slapped an "in order of preference" at the end of the second sentence

**Pull request merged**